---
title: "Henry County Mitigation Project 2013: Microbial Community Charactorization"
author: "Ariane L. Peralta, Mario E. Muscarella, Jeffrey W. Matthews"
date: "Last updated on `r format(Sys.time(), '%d %B, %Y')`"
header-includes:
  - \usepackage{array}
  - \usepackage{graphics}
output: 
  pdf_document:
  fig_caption: true
---

Project Description: 

# Initial Setup
```{r}
rm(list=ls())
setwd("~/GitHub/IL_Wetlands/analyses")
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
ci <- function(x, ...){1.96 * sd(x,na.rm = TRUE)}

# Code Dependencies
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")
require("vegan")
require("reshape")
require("ggplot2")
require("nlme")
require("ade4")
require("grid"); require("png")
require("ape"); require("picante")

# Define Colors
#myColors <- c("#FFF000", "#CCFF00", "#33CC33", "#000000", "#336633", "#FF9933") #pick new
myColors <- c("#448844", "#33CC33", "#CCFF00", "#FFF000", "#FF9933", "#A9A9A9")
names(myColors) <- c("BalledBurlapped", "Bareroot", "Seedling", "Acorn", "Seedbank", "Reference")
```

# Microbial Data
```{r}
# Import OTU data
# Import Raw Data
WLdata.in <- read.otu("../data/WL.final.shared")

# Remove Mock Community
WLdata.in2 <- WLdata.in[which(rownames(WLdata.in) != "Mock"), ]

# Remove OTUs with less than two occurences across all sites
WLdata <- WLdata.in2[, which(colSums(WLdata.in2) >= 2)]

# Make Presence Absence Matrix
WLdataPA <- (WLdata > 0) * 1

# Make Relative Abundence Matrices
WLdataREL <- WLdata
for(i in 1:dim(WLdata)[1]){
  WLdataREL[i,] <- WLdata[i,]/sum(WLdata[i,])
}

# Log Transform Relative Abundances
WLdataREL.log <- decostand(WLdataREL, method="log")

# Import Taxonomy File
WL.tax <- read.tax(taxonomy = "../data/WL.final.0.03.taxonomy")
```

# Plant Data
```{r}
WL.plant <- read.csv("../data/IL_Wetlands_Plants/IL_Wetlands_HC_Plants.csv", row.names = 1)

# The plant data is already relativized
# But it might be useful to have it as PA

WL.plant.PA <- (WL.plant > 0) * 1
```

# Phylogenetic Analysis
```{sh}
# The following was done outside of R

python ../bin/name_change.py WL.final.0.03.rep.fasta WL.final.0.03.rep.rename.fasta

FastTree -gtr -nt -gamma -fastest WL.final.0.03.rep.rename.fasta > WL.bac.tree 

Output: 
ML-NNI round 11: LogLk = -1017514.896 NNIs 4825 max delta 3.21 Time 626.69 (final)
Optimize all lengths: LogLk = -1017490.876 Time 645.88 
Gamma(20) LogLk = -1017848.611 alpha = 2.130 rescaling lengths by 1.471   
Total time: 733.31 seconds Unique: 56413/56413 Bad splits: 16/56410 Worst delta-LogLk 3.347

# Weighted UniFrac was done using Mothur

This caused an error because of the names. Mothur actually crashed

FastTree -gtr -nt -gamma -fastest WL.final.0.03.rep.fasta > WL.bac.tree

ML-NNI round 12: LogLk = -1017096.874 NNIs 4848 max delta 3.23 Time 633.03 (final)x delta 3.226)   
Optimize all lengths: LogLk = -1017090.230 Time 650.74 
Gamma(20) LogLk = -1017412.647 alpha = 2.058 rescaling lengths by 1.484   
Total time: 735.70 seconds Unique: 56413/56413 Bad splits: 13/56410 Worst delta-LogLk 1.738

Mothur (v 1.38)
unifrac.weighted(tree=WL.bac.tree, count=WL.final.rep.count_table, distance=square)

Output File Names: 
WL.bac.treewsummary
WL.bac.tree1.weighted.phylip.dist
```

## Import Phylogenetic Tree and UniFrac Distance Matrix
```{r}
#WL.phylo <- read.tree("../data/WL.bac.renamed.tree")

WL.unifrac.raw <- read.delim("../data/WL.bac.tree1.weighted.phylip.dist", header = F, skip = 1, row.names = 1)
WL.unifrac <- WL.unifrac.raw[which(row.names(WL.unifrac.raw) %in% row.names(WLdata)), which(row.names(WL.unifrac.raw) %in% row.names(WLdata))]
WL.unifrac.dist <- as.dist(WL.unifrac, upper = T, diag = T)
```

# Soil Data
```{r}
WL.soil <- read.csv("../data/WL_plant_soil.csv")
row.names(WL.soil) <- WL.soil$Sample_Code

WL.soil$Treatment <- factor(WL.soil$Treatment, 
                           levels = c("BalledBurlapped", "Bareroot", 
                           "Seedling", "Acorn", "Seedbank", "Reference"))
```

# Environmental Data
```{r}
# Import Environmental Data
design <- read.csv("../data/WL.design.csv", row.names = 1)
design$Treatment <- factor(design$Treatment, 
                           levels = c("BalledBurlapped", "Bareroot", 
                           "Seedling", "Acorn", "Seedbank", "Reference"))
treatments <- design$Treatment 
#levels(treatments) == c("BalledBurlapped", "Bareroot", "Seedling", "Acorn", 
#                        "Seedbank", "Reference")
```

# Simple Hypothesis Testing
```{r}
# Check that Bac and Plant data have same structure as design
row.names(WLdataREL) == row.names(design)
row.names(WL.unifrac) == row.names(design)
row.names(WL.plant) == row.names(design) 
  # Same info seem to be inside so I'm going to rename the plant data
row.names(WL.plant) <- row.names(design) 

# PERMANOVA
WL.bac.adonis <- adonis(WLdataREL ~ treatments, method = "bray", perm=999)
WL.bac.adonis

# Odd sites in bacterial composition data (explore more)
odd.sites <- c("HC_A3_ss1_S11", "HC_A3_ss2_S12", "HC_A3_ss3_S13", "HC_C3_ss5_S45", "HC_D2_ss3_S53", "HC_E1_ss1_S61")

WLdataREL.2 <- WLdataREL[setdiff(rownames(WLdataREL), odd.sites), ]
treatments.2 <- design[setdiff(rownames(design), odd.sites), ]$Treatment

WL.bac.adonis <- adonis(WLdataREL.2 ~ treatments.2, method = "bray", perm=999)
WL.bac.adonis

WL.unifrac.adonis <- adonis(WL.unifrac.dist ~ treatments, perm = 999)
WL.unifrac.adonis

WL.plant.adonis <- adonis(WL.plant ~ treatments, method = "bray", perm=999)
WL.plant.adonis

# Simper Analysis for Species Responses
#WL.bac.simper <- simper(WLdataREL, group = treatments, permutations = 999)
#bac.sum <- summary(WL.bac.simper, ordered = T)

WL.plant.simper <- simper(WL.plant, group = treatments, permutations = 999)
plant.sum <- summary(WL.plant.simper, ordered = T)
plant.sum
```

# Microbial Bray Curtis Ordination (PCoA)
```{r}
png(filename="../figures/WL.bac.PCoA.png",
    width = 1200, height = 800, res = 96*2)
# Create Distance Matrix
sampleREL.dist <- vegdist(WLdataREL, method="bray")

# Principal Coordinates Analysis
WL_pcoa <- cmdscale(sampleREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

# Remove Odd Sites
odd.sites <- row.names(WLdataREL[c(abs(WL_pcoa$points[, 1]) > 0.3), ])
odd.sbys <- WLdataREL[c(abs(WL_pcoa$points[, 1]) > 0.3), ]
mean(rowSums((WLdataREL > 0) * 1))
rowSums((odd.sbys > 0) * 1)
WLdataREL.2 <- WLdataREL[c(abs(WL_pcoa$points[, 1]) < 0.3), ]
design2 <- design[c(abs(WL_pcoa$points[, 1]) < 0.3), ]
treatments <- design2$Treatment

# Create Distance Matrix
sampleREL.dist2 <- vegdist(WLdataREL.2, method="bray")

# Principal Coordinates Analysis
WL_pcoa <- cmdscale(sampleREL.dist2, k=2, eig=TRUE, add=FALSE)
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1 <- round(WL_pcoa$eig[1] / sum(WL_pcoa$eig), 3) * 100
explainvar2 <- round(WL_pcoa$eig[2] / sum(WL_pcoa$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2)

# Plot
points <- cbind(as.data.frame(WL_pcoa$points), treatments)
L.centroids <- melt(points, id="treatments", measure.vars = c("V1", "V2"))
centroids <- cast(L.centroids, variable ~ treatments, mean)
centroids.se <- cast(L.centroids, variable ~ treatments, se)
centroids.sd <- cast(L.centroids, variable ~ treatments, sd)

cent.dataframe <- t(data.frame(rbind(centroids[1,-1], centroids[2,-1],
                             centroids.sd[1,-1],centroids.sd[2,-1])))
colnames(cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
cent.treats <- rownames(cent.dataframe)

# Define Plot Parameters
par(mar = c(5, 5.5, 1, 1) + 0.1)
layout(matrix(1:2, 1, 2), widths = c(4,2))

plot(cent.dataframe[,1], cent.dataframe[,2], type = 'n', las = 1,
     xlim = c(-0.1, 0.1), ylim = c(-0.1, 0.1),
     xaxt = "n", xlab = "", yaxt = "n", ylab="")

abline(h = 0, lty = 3, col = "gray")
abline(v = 0, lty = 3, col = "gray")

arrows(x0 = cent.dataframe[,1], 
       y1 = cent.dataframe[,2] - cent.dataframe[,4], 
       y0 = cent.dataframe[,2] + cent.dataframe[,4],
       angle = 90,
       length=0.1, lwd = 2, code = 3)
arrows(y0 = cent.dataframe[,2], 
       x1 = cent.dataframe[,1] - cent.dataframe[,3], 
       x0 = cent.dataframe[,1] + cent.dataframe[,3],
       angle = 90,
       length=0.1, lwd = 2, code = 3)
points(cent.dataframe[,1], cent.dataframe[,2], 
       cex = 2.5, bg = myColors, col = "black", pch = 22, lwd = 2)

# text(cent.dataframe[,1], cent.dataframe[,2], rownames(cent.dataframe))

axis(side = 1, labels = T, las = 1, lwd.ticks = 2)
axis(side = 2, labels = T, las = 1, lwd.ticks = 2)
axis(side=1, lwd.ticks = 2, tck = -0.02, labels=F, cex.axis=1)
axis(side=3, lwd.ticks = 2, tck = -0.02, labels=F, cex.axis=1)
axis(side=1, lwd.ticks = 2, tck = 0.01, labels=F, cex.axis=1)
axis(side=3, lwd.ticks = 2, tck = 0.01, labels=F, cex.axis=1)
axis(side = 2, lwd.ticks = 2, tck = -0.02, labels=F, cex.axis=1)
axis(side = 4, lwd.ticks = 2, tck = -0.02, labels=F, cex.axis=1)
axis(side = 2, lwd.ticks = 2, tck = 0.01, labels=F, cex.axis=1)
axis(side = 4, lwd.ticks = 2, tck = 0.01, labels=F, cex.axis=1)

mtext(paste("PCoA 1 (", explainvar1, "%)", sep = ""), side = 1, line = 3, cex = 1.5)
mtext(paste("PCoA 2 (", explainvar2, "%)", sep = ""), side = 2, line = 3.5, cex = 1.5)
      
box(lwd = 2)

par(mar = c(5, 0, 1, 1) + 0.1)
plot.new()

legend(0, 1, legend = levels(treatments), pt.bg = myColors, col = "black",
       pch = 22, cex = 0.9, bty = 'n', inset = c(0.1, 0.05),
       y.intersp = 1.25)


# Close Plot Device
dev.off()
graphics.off()
#```
## Show Plot
#```{r}
img <- readPNG("../figures/WL.bac.PcoA.png")
grid.raster(img)
```

## Microbial Phylogenetic Ordination 
```{r}
png(filename="../figures/WL.unifrac.PCoA.png",
    width = 1200, height = 800, res = 96*2)

# Principal Coordinates Analysis
WL_pcoa <- cmdscale(WL.unifrac.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

# Remove Odd Sites
odd.sites <- row.names(WL.unifrac[c(abs(WL_pcoa$points[, 1]) > 0.1), ])


# Create Distance Matrix W/O Odd Sites
WL.unifrac.2 <- WL.unifrac[-c(which(row.names(WL.unifrac) %in% odd.sites)),
                           -c(which(row.names(WL.unifrac) %in% odd.sites))]
WL.unifrac.dist2 <- as.dist(WL.unifrac.2, upper = T, diag = T)

treatments <- design[-c(which(row.names(design) %in% odd.sites)), ]$Treatment


# Principal Coordinates Analysis
WL_pcoa <- cmdscale(WL.unifrac.dist2, k=2, eig=TRUE, add=FALSE)
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1 <- round(WL_pcoa$eig[1] / sum(WL_pcoa$eig), 3) * 100
explainvar2 <- round(WL_pcoa$eig[2] / sum(WL_pcoa$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2)

# Plot
points <- cbind(as.data.frame(WL_pcoa$points), treatments)
L.centroids <- melt(points, id="treatments", measure.vars = c("V1", "V2"))
centroids <- cast(L.centroids, variable ~ treatments, mean)
centroids.se <- cast(L.centroids, variable ~ treatments, se)
centroids.sd <- cast(L.centroids, variable ~ treatments, sd)

cent.dataframe <- t(data.frame(rbind(centroids[1,-1], centroids[2,-1],
                             centroids.sd[1,-1],centroids.sd[2,-1])))
colnames(cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
cent.treats <- rownames(cent.dataframe)

# Define Plot Parameters
par(mar = c(5, 5.5, 1, 1) + 0.1)
layout(matrix(1:2, 1, 2), widths = c(4,2))

plot(cent.dataframe[,1], cent.dataframe[,2], type = 'n', las = 1,
     xlim = c(-0.1, 0.1), ylim = c(-0.1, 0.1),
     xaxt = "n", xlab = "", yaxt = "n", ylab="")

abline(h = 0, lty = 3, col = "gray")
abline(v = 0, lty = 3, col = "gray")

arrows(x0 = cent.dataframe[,1], 
       y1 = cent.dataframe[,2] - cent.dataframe[,4], 
       y0 = cent.dataframe[,2] + cent.dataframe[,4],
       angle = 90,
       length=0.1, lwd = 2, code = 3)
arrows(y0 = cent.dataframe[,2], 
       x1 = cent.dataframe[,1] - cent.dataframe[,3], 
       x0 = cent.dataframe[,1] + cent.dataframe[,3],
       angle = 90,
       length=0.1, lwd = 2, code = 3)
points(cent.dataframe[,1], cent.dataframe[,2], 
       cex = 2.5, bg = myColors, col = "black", pch = 22, lwd = 2)

#text(cent.dataframe[,1], cent.dataframe[,2], rownames(cent.dataframe))

axis(side = 1, labels = T, las = 1, lwd.ticks = 2)
axis(side = 2, labels = T, las = 1, lwd.ticks = 2)
axis(side=1, lwd.ticks = 2, tck = -0.02, labels=F, cex.axis=1)
axis(side=3, lwd.ticks = 2, tck = -0.02, labels=F, cex.axis=1)
axis(side=1, lwd.ticks = 2, tck = 0.01, labels=F, cex.axis=1)
axis(side=3, lwd.ticks = 2, tck = 0.01, labels=F, cex.axis=1)
axis(side = 2, lwd.ticks = 2, tck = -0.02, labels=F, cex.axis=1)
axis(side = 4, lwd.ticks = 2, tck = -0.02, labels=F, cex.axis=1)
axis(side = 2, lwd.ticks = 2, tck = 0.01, labels=F, cex.axis=1)
axis(side = 4, lwd.ticks = 2, tck = 0.01, labels=F, cex.axis=1)

mtext(paste("PCoA 1 (", explainvar1, "%)", sep = ""), side = 1, line = 3, cex = 1.5)
mtext(paste("PCoA 2 (", explainvar2, "%)", sep = ""), side = 2, line = 3.5, cex = 1.5)
      
box(lwd = 2)

par(mar = c(5, 0, 1, 1) + 0.1)
plot.new()

legend(0, 1, legend = cent.treats, pt.bg = myColors, col = "black",
       pch = 22, cex = 0.9, bty = 'n', inset = c(0.1, 0.05),
       y.intersp = 1.25)


# Close Plot Device
dev.off()
graphics.off()
#```
## Show Plot
#```{r}
img <- readPNG("../figures/WL.unifrac.PcoA.png")
grid.raster(img)
```


# Plant Ordinations
## Principal Coordinates Ordination
```{r}
png(filename="../figures/WL.plant.PCoA.png",
    width = 1200, height = 800, res = 96*2)
# Create Distance Matrix
treatments <- design$Treatment
samplePlant.dist <- vegdist(WL.plant, method="bray")

# Principal Coordinates Analysis
WLplant_pcoa <- cmdscale(samplePlant.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

# # Remove Odd Sites
# odd.sites <- row.names(WLdataREL[c(abs(WL_pcoa$points[, 1]) > 0.3), ])
# odd.sbys <- WLdataREL[c(abs(WL_pcoa$points[, 1]) > 0.3), ]
# mean(rowSums((WLdataREL > 0) * 1))
# rowSums((odd.sbys > 0) * 1)
# WLdataREL.2 <- WLdataREL[c(abs(WL_pcoa$points[, 1]) < 0.3), ]
# design2 <- design[c(abs(WL_pcoa$points[, 1]) < 0.3), ]
# treatments <- design2$Treatment

# Create Distance Matrix
#sampleREL.dist2 <- vegdist(WLdataREL.2, method="bray")

# Principal Coordinates Analysis
#WL_pcoa <- cmdscale(sampleREL.dist2, k=2, eig=TRUE, add=FALSE)
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1 <- round(WLplant_pcoa$eig[1] / sum(WLplant_pcoa$eig), 3) * 100
explainvar2 <- round(WLplant_pcoa$eig[2] / sum(WLplant_pcoa$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2)

# Plot
points <- cbind(as.data.frame(WLplant_pcoa$points), treatments)
L.centroids <- melt(points, id="treatments", measure.vars = c("V1", "V2"))
centroids <- cast(L.centroids, variable ~ treatments, mean)
centroids.se <- cast(L.centroids, variable ~ treatments, se)
centroids.sd <- cast(L.centroids, variable ~ treatments, sd)

cent.dataframe <- t(data.frame(rbind(centroids[1,-1], centroids[2,-1],
                             centroids.sd[1,-1],centroids.sd[2,-1])))
colnames(cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
cent.treats <- rownames(cent.dataframe)

# Define Plot Parameters
par(mar = c(5, 5.5, 1, 1) + 0.1)
layout(matrix(1:2, 1, 2), widths = c(4,2))

plot(cent.dataframe[,1], cent.dataframe[,2], type = 'n', las = 1,
     xlim = c(-0.8, 0.8), ylim = c(-0.8, 0.8),
     xaxt = "n", xlab = "", yaxt = "n", ylab="")

abline(h = 0, lty = 3, col = "gray")
abline(v = 0, lty = 3, col = "gray")

arrows(x0 = cent.dataframe[,1], 
       y1 = cent.dataframe[,2] - cent.dataframe[,4], 
       y0 = cent.dataframe[,2] + cent.dataframe[,4],
       angle = 90,
       length=0.1, lwd = 2, code = 3)
arrows(y0 = cent.dataframe[,2], 
       x1 = cent.dataframe[,1] - cent.dataframe[,3], 
       x0 = cent.dataframe[,1] + cent.dataframe[,3],
       angle = 90,
       length=0.1, lwd = 2, code = 3)
points(cent.dataframe[,1], cent.dataframe[,2], 
       cex = 2.5, bg = myColors, col = "black", pch = 22, lwd = 2)

axis(side = 1, labels = T, las = 1, lwd.ticks = 2)
axis(side = 2, labels = T, las = 1, lwd.ticks = 2)
axis(side=1, lwd.ticks = 2, tck = -0.02, labels=F, cex.axis=1)
axis(side=3, lwd.ticks = 2, tck = -0.02, labels=F, cex.axis=1)
axis(side=1, lwd.ticks = 2, tck = 0.01, labels=F, cex.axis=1)
axis(side=3, lwd.ticks = 2, tck = 0.01, labels=F, cex.axis=1)
axis(side = 2, lwd.ticks = 2, tck = -0.02, labels=F, cex.axis=1)
axis(side = 4, lwd.ticks = 2, tck = -0.02, labels=F, cex.axis=1)
axis(side = 2, lwd.ticks = 2, tck = 0.01, labels=F, cex.axis=1)
axis(side = 4, lwd.ticks = 2, tck = 0.01, labels=F, cex.axis=1)

mtext(paste("PCoA 1 (", explainvar1, "%)", sep = ""), side = 1, line = 3, cex = 1.5)
mtext(paste("PCoA 2 (", explainvar2, "%)", sep = ""), side = 2, line = 3.5, cex = 1.5)
      
box(lwd = 2)

par(mar = c(5, 0, 1, 1) + 0.1)
plot.new()

legend(0, 1, legend = cent.treats, pt.bg = myColors, col = "black", 
       pch = 22, cex = 0.9, bty = 'n', inset = c(0.1, 0.05),
       y.intersp = 1.25)


# Close Plot Device
dev.off()
graphics.off()
#```
## Show Plot
#```{r}
img <- readPNG("../figures/WL.plant.PcoA.png")
grid.raster(img)
```


# Similarity between microbe and plant communities
```{r}
# Name check
rownames(WL.plant) == rownames(WLdataREL)

# Remove Odd Sites from Bac Community
bac.comm <- WLdataREL[-which(rownames(WLdataREL) %in% odd.sites),]
plant.comm <- WL.plant[-which(rownames(WL.plant) %in% odd.sites), ]

# Remove Bare Treatment
bare.sites <- rownames(design[design$Treatment == "Bareroot", ])
bac.comm <- bac.comm[-which(rownames(bac.comm) %in% bare.sites),]
plant.comm <- plant.comm[-which(rownames(plant.comm) %in% bare.sites), ]

rownames(bac.comm) == rownames(plant.comm)

rowSums(plant.comm)

dist.bac <- vegdist(bac.comm, method = "bray")
dist.plant <- vegdist(plant.comm, method = "bray")

mantel.rtest(dist.bac, dist.plant)

WL.plant_points <- cmdscale(dist.plant, k=3, eig=TRUE, add=FALSE)$points

dbRDA <- dbrda(bac.comm ~ WL.plant_points, distance = "bray")
anova(dbRDA) 

RsquareAdj(dbRDA)
```

# Soil Factors
```{r}

head(WL.soil)
# Impt Soil: Moisture, Temp, pH, TOC, TN, OM, NH4, NO3

WL.trts <- WL.soil[, "Treatment"]
WL.soil.phys <- WL.soil[, which(colnames(WL.soil) %in% c("Moisture", "Temp", "pH"))]
WL.soil.nuts <- WL.soil[, which(colnames(WL.soil) %in% c("TOC", "TN", "OM", "NH4.N", "NO3.N"))]

library(agricolae)

moisture.lm <- lm(Moisture ~ Treatment, data=WL.soil)
anova(moisture.lm)
summary(moisture.lm)
HSD <- HSD.test(moisture.lm,"Treatment", console=TRUE)

pH.lm <- lm(pH ~ Treatment, data=WL.soil)
anova(pH.lm)
summary(pH.lm)
HSD <- HSD.test(pH.lm,"Treatment", console=TRUE)

temp.lm <- lm(Temp ~ Treatment, data=WL.soil)
anova(temp.lm)
summary(temp.lm)
HSD <- HSD.test(temp.lm,"Treatment", console=TRUE)

TOC.lm <- lm(TOC ~ Treatment, data=WL.soil)
anova(TOC.lm)
summary(TOC.lm)
HSD <- HSD.test(TOC.lm,"Treatment", console=TRUE)

NH4.lm <- lm(NH4.N ~ Treatment, data=WL.soil)
anova(NH4.lm)
summary(NH4.lm)

NO3.lm <- lm(NO3.N ~ Treatment, data=WL.soil)
anova(NO3.lm)
summary(NO3.lm)
HSD <- HSD.test(NO3.lm,"Treatment", console=TRUE)

WL.phys.mean <- apply(WL.soil.phys, 2, 
                      FUN = function(avg) aggregate(avg ~ WL.trts, WL.soil.phys, mean))
WL.phys.sem <- apply(WL.soil.phys, 2, 
                     FUN = function(sem) aggregate(sem ~ WL.trts, WL.soil.phys, se))
WL.phys.95 <- apply(WL.soil.phys, 2, 
                    FUN = function(ci_95) aggregate(ci_95 ~ WL.trts, WL.soil.phys, ci))

WL.nuts.mean <- apply(WL.soil.nuts, 2, 
                      FUN = function(avg) aggregate(avg ~ WL.trts, WL.soil.nuts, mean))
WL.nuts.sem <- apply(WL.soil.nuts, 2, 
                     FUN = function(sem) aggregate(sem ~ WL.trts, WL.soil.nuts, se))
WL.nuts.95 <- apply(WL.soil.nuts, 2, 
                    FUN = function(ci_95) aggregate(ci_95 ~ WL.trts, WL.soil.nuts, ci))


phys.means.table <- data.frame(trt = WL.phys.mean$Moisture$WL.trts,
                               moisture = WL.phys.mean$Moisture$avg,
                               temp = WL.phys.mean$Temp$avg,
                               pH = WL.phys.mean$pH$avg)
phys.sem.table <- data.frame(trt = WL.phys.sem$Moisture$WL.trts,
                               moisture = WL.phys.sem$Moisture$sem,
                               temp = WL.phys.sem$Temp$sem,
                               pH = WL.phys.sem$pH$sem)
phys.ci.table <- data.frame(trt = WL.phys.95$Moisture$WL.trts,
                               moisture = WL.phys.95$Moisture$ci_95,
                               temp = WL.phys.95$Temp$ci_95,
                               pH = WL.phys.95$pH$ci_95)

nuts.means.table <- data.frame(trt = WL.nuts.mean$TOC$WL.trts,
                               toc = WL.nuts.mean$TOC$avg,
                               tn = WL.nuts.mean$TN$avg,
                               om = WL.nuts.mean$OM$avg,
                               nh4 = WL.nuts.mean$NH4.N$avg,
                               no3 = WL.nuts.mean$NO3.N$avg)
nuts.sem.table <- data.frame(trt = WL.nuts.sem$TOC$WL.trts,
                               toc = WL.nuts.sem$TOC$sem,
                               tn = WL.nuts.sem$TN$sem,
                               om = WL.nuts.sem$OM$sem,
                               nh4 = WL.nuts.sem$NH4.N$sem,
                               no3 = WL.nuts.sem$NO3.N$sem)
nuts.ci.table <- data.frame(trt = WL.nuts.95$TOC$WL.trts,
                               toc = WL.nuts.95$TOC$ci_95,
                               tn = WL.nuts.95$TN$ci_95,
                               om = WL.nuts.95$OM$ci_95,
                               nh4 = WL.nuts.95$NH4.N$ci_95,
                               no3 = WL.nuts.95$NO3.N$ci_95)
```

# WL Soil Plots
```{r}
png(filename="../figures/soil.chem.png",
    width = 800, height = 1200, res = 96*2)

layout(matrix(1:3, 3, byrow = T))
par(mar = (c(0.5, 5, 0, 1) + 0.1), oma = c(7, 1, 1.5, 1))

# TOC
boxplot(WL.soil.nuts$TOC ~ WL.trts, 
        col = myColors, xaxt = "n",  yaxt = "n",
        xlab = "", ylab = "", ylim = c(4,6))

  # Axes with Tick Marks
  axis(side = 1, labels = F, tck = -0.01, lwd = 2)
  axis(side = 2, labels = T, tck = -0.02, lwd = 2, las = 1)
  axis(side = 4, labels = F, tck = -0.02, lwd = 2)
  axis(side = 2, labels = F, tck = 0.01, lwd = 2)
  axis(side = 4, labels = F, tck = 0.01, lwd = 2)
  box(lwd = 2)
  
  # Lables
  mtext("Total Organic Carbon\n(%)", side = 2, cex = 1, line = 3.5)

# # Total Nitrogen
# boxplot(WL.soil.nuts$TN ~ WL.trts, 
#         col = myColors, xaxt = "n",  yaxt = "n",
#         xlab = "", ylab = "", ylim = c(0.25, 0.4))
# 
#   # Axes with Tick Marks
#   axis(side = 1, labels = F, tck = -0.01, lwd = 2)
#   axis(side = 2, labels = T, tck = -0.02, lwd = 2, las = 1)
#   axis(side = 4, labels = F, tck = -0.02, lwd = 2)
#   axis(side = 2, labels = F, tck = 0.01, lwd = 2)
#   axis(side = 4, labels = F, tck = 0.01, lwd = 2)
#   box(lwd = 2)
#   
#   # Lables
#   mtext("Total Nitrogen\n(?)", side = 2, cex = 1, line = 3.5)

# # Organic Matter
# boxplot(WL.soil.nuts$OM ~ WL.trts, 
#         col = myColors, xaxt = "n",  yaxt = "n",
#         xlab = "", ylab = "", ylim = c(7, 11))
# 
#   # Axes with Tick Marks
#   axis(side = 1, labels = F, tck = -0.01, lwd = 2)
#   axis(side = 2, labels = T, tck = -0.02, lwd = 2, las = 1)
#   axis(side = 4, labels = F, tck = -0.02, lwd = 2)
#   axis(side = 2, labels = F, tck = 0.01, lwd = 2)
#   axis(side = 4, labels = F, tck = 0.01, lwd = 2)
#   box(lwd = 2)
#   
#   # Lables
#   mtext("Organic Matter\n(?)", side = 2, cex = 1, line = 3.5)

# Ammonium
boxplot(WL.soil.nuts$NH4.N ~ WL.trts, 
        col = myColors, xaxt = "n",  yaxt = "n",
        xlab = "", ylab = "", ylim = c(6, 14))

  # Axes with Tick Marks
  axis(side = 1, labels = F, tck = -0.01, lwd = 2)
  axis(side = 2, labels = T, tck = -0.02, lwd = 2, las = 1)
  axis(side = 4, labels = F, tck = -0.02, lwd = 2)
  axis(side = 2, labels = F, tck = 0.01, lwd = 2)
  axis(side = 4, labels = F, tck = 0.01, lwd = 2)
  box(lwd = 2)
  
  # Labels
  mtext("Ammonium\n(mg/kg)", side = 2, cex = 1, line = 3.5)

# Nitrate
boxplot(WL.soil.nuts$NO3.N ~ WL.trts, 
        col = myColors, xaxt = "n",  yaxt = "n",
        xlab = "", ylab = "")

  # Axes with Tick Marks
  axis(side = 1, labels = F, tck = -0.01, lwd = 2)
  axis(side = 2, labels = T, tck = -0.02, lwd = 2, las = 1)
  axis(side = 4, labels = F, tck = -0.02, lwd = 2)
  axis(side = 2, labels = F, tck = 0.01, lwd = 2)
  axis(side = 4, labels = F, tck = 0.01, lwd = 2)
  box(lwd = 2)
  
  # Lables
  mtext("Nitrate\n(mg/kg)", side = 2, cex = 1, line = 3.5)

# Plot X labs at default X position
par(xpd = NA)
text(x = seq_along(levels(WL.trts)), y = par("usr")[3] - 0.1 * (par("usr")[4] - par("usr")[3]), srt = 45, adj = 1, labels = levels(WL.trts), xpd=NA)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("../figures/soil.chem.png")
grid.raster(img)
```


# WL Soil Plots
```{r}
png(filename="../figures/soil.phy.png",
    width = 800, height = 1200, res = 96*2)


layout(matrix(1:3, 3, byrow = T))
par(mar = (c(0.5, 5, 0, 1) + 0.1), oma = c(7, 1, 1.5, 1))

boxplot(WL.soil.phys$Moisture ~ WL.trts, 
        col = myColors, xaxt = "n",  yaxt = "n",
        xlab = "", ylab = "", ylim = c(20, 50))

  # Axes with Tick Marks
  axis(side = 1, labels = F, tck = -0.01, lwd = 2)
  axis(side = 2, labels = T, tck = -0.02, lwd = 2, las = 1)
  axis(side = 4, labels = F, tck = -0.02, lwd = 2)
  axis(side = 2, labels = F, tck = 0.01, lwd = 2)
  axis(side = 4, labels = F, tck = 0.01, lwd = 2)
  box(lwd = 2)

  # Lables
  mtext("Moisture (%)", side = 2, cex = 1, line = 3.5)

boxplot(WL.soil.phys$Temp ~ WL.trts, 
        col = myColors, xaxt = "n",  yaxt = "n",
        xlab = "", ylab = "", ylim = c(18, 26))

  # Axes with Tick Marks
  axis(side = 1, labels = F, tck = -0.01, lwd = 2)
  axis(side = 2, labels = T, tck = -0.02, lwd = 2, las = 1)
  axis(side = 4, labels = F, tck = -0.02, lwd = 2)
  axis(side = 2, labels = F, tck = 0.01, lwd = 2)
  axis(side = 4, labels = F, tck = 0.01, lwd = 2)
  box(lwd = 2)

  # Lables
  mtext("Temperature (ºC)", side = 2, cex = 1, line = 3.5)

boxplot(WL.soil.phys$pH ~ WL.trts, 
        col = myColors, xaxt = "n",  yaxt = "n",
        xlab = "", ylab = "", ylim = c(7.2, 7.8))

  # Axes with Tick Marks
  axis(side = 1, labels = F, tck = -0.01, lwd = 2)
  axis(side = 2, labels = T, tck = -0.02, lwd = 2, las = 1)
  axis(side = 4, labels = F, tck = -0.02, lwd = 2)
  axis(side = 2, labels = F, tck = 0.01, lwd = 2)
  axis(side = 4, labels = F, tck = 0.01, lwd = 2)
  box(lwd = 2)

  # Lables
  mtext("pH", side = 2, cex = 1, line = 3.5)


# Plot x labs at default x position
par(xpd = NA)
text(x = seq_along(levels(WL.trts)), y = par("usr")[3] - 0.1 * (par("usr")[4] - par("usr")[3]), srt = 45, adj = 1, labels = levels(WL.trts), xpd=NA)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("../figures/soil.phy.png")
grid.raster(img)
```










# Distance Based RDA Models
```{r}
phys.dist <- dist(WL.soil.phys, method = "euclidian")
phys.pcoa <- cmdscale(phys.dist, k = 3, eig = T)

explainvar1 <- round(phys.pcoa$eig[1] / sum(phys.pcoa$eig), 3) * 100
explainvar2 <- round(phys.pcoa$eig[2] / sum(phys.pcoa$eig), 3) * 100

chem.dist <- dist(WL.soil.nuts, method = "euclidian")
chem.pcoa <- cmdscale(chem.dist, k = 3, eig = T)

explainvar1 <- round(chem.pcoa$eig[1] / sum(chem.pcoa$eig), 3) * 100
explainvar2 <- round(chem.pcoa$eig[2] / sum(chem.pcoa$eig), 3) * 100

dbRDA <- dbrda(WLdataREL ~ chem.pcoa$points[,1], distance = "bray")
anova(dbRDA) 

dbRDA <- dbrda(WLdataREL ~ phys.pcoa$points[,1], distance = "bray")
anova(dbRDA)
RsquareAdj(dbRDA)

dbRDA <- dbrda(WLdataREL ~ phys.pcoa$points[,1] * chem.pcoa$points[,1], 
               distance = "bray", add = T)
anova(dbRDA, by = "terms", model = "direct")
RsquareAdj(dbRDA)
summary(dbRDA)

dbRDA <- dbrda(WL.plant ~ phys.pcoa$points[,1] * chem.pcoa$points[,1], 
               distance = "bray", add = T)
anova(dbRDA, by = "terms", model = "direct") 
RsquareAdj(dbRDA)
```










```{r}
phys.mod <- lm(phys.pcoa$points[,1] ~ WL.trts)
chem.mod <- lm(chem.pcoa$points[,1] ~ WL.trts)
anova(phys.mod)
anova(chem.mod)

TukeyHSD(aov(phys.mod))
TukeyHSD(aov(chem.mod))

boxplot(phys.pcoa$points[,1] ~ WL.trts)
boxplot(chem.pcoa$points[,1] ~ WL.trts)
```



# Invasive Species Biomass
```{r}
WL.rc <- data.frame(WL.soil$Treatment, WL.soil$RCGbiomass, WL.soil$Non.RCGbiomass)

WL.rc$ratio <- WL.rc$WL.soil.RCGbiomass / (WL.rc$WL.soil.RCGbiomass + WL.rc$WL.soil.Non.RCGbiomass) 
```


```{r}
png(filename="../figures/reed.biomass.png",
    width = 1200, height = 800, res = 96*2)


par(mar = (c(6.5,5,1,1) + 0.1))

boxplot(WL.rc$WL.soil.RCGbiomass ~ WL.rc$WL.soil.Treatment, 
        col = myColors, xaxt = "n",  yaxt = "n",
        xlab = "", ylab = "")

# Lables
mtext("Reed Canary Biomass (g)", side = 2, cex = 1.25, line = 3)

# Plot x labs at default x position
text(x = seq_along(levels(WL.rc$WL.soil.Treatment)), 
     y = par("usr")[3] - 10, srt = 45, adj = 1, 
     labels = levels(WL.rc$WL.soil.Treatment), 
     xpd = TRUE)

# Axes with Tick Marks
axis(side = 1, labels = F, tck = -0.01, lwd = 2)
axis(side = 2, labels = T, tck = -0.02, lwd = 2, las = 1)
#axis(side = 3, labels = F, tck = -0.02, lwd = 2)
axis(side = 4, labels = F, tck = -0.02, lwd = 2)
#axis(side = 1, labels = F, tck = 0.01, lwd = 2)
axis(side = 2, labels = F, tck = 0.01, lwd = 2)
#axis(side = 3, labels = F, tck = 0.01, lwd = 2)
axis(side = 4, labels = F, tck = 0.01, lwd = 2)

box(lwd = 2)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("../figures/reed.biomass.png")
grid.raster(img)
```




##############
OLD CODE
##############

## Responder Analysis Based on PCoA
```{r}
# pcoaS <- add.spec.scores(WL_pcoa, WLdataREL, method="cor.scores",Rscale=TRUE,
#   scaling=1,multi=1)
  # retrieves correlation coefficient for each taxon's relative
  # abudnace with respect to PCoA coordinates (k = 3)

pcoaS <- wascores(WL_pcoa$points, WLdataREL, expand=T)
# pcoaS <- pcoascores(WL_pcoa$points, WLdataREL, expand=T)
# eigengrad(WL_pcoa$points, WLdataREL)

# cor(WL_pcoa, pcoaS)

# Species Correlations

# ### BEGIN R
# ## NMDS
# library(vegan)
# data(dune)
# m <- metaMDS(dune)
# cor(dune, scores(m, dis="si"))
# ## capscale, copying the previous method, but 4 axes
# m <- capscale(dune ~ 1)
# cor(dune, scores(m, dis="si", choices=1:4))
#     ## capscale, the direct way re-using the previous result
#     scores(m, dis="sp", scaling=-2, const = sqrt(nrow(dune)-1)))
# ### END
```

# Mario's Stopping Point 1/13/16

# # PCoA Axis 1 Responders
# cor_spp_a1 <- cbind(EC_tax[,3],pcoaS$cproj[,1])
#   # creates matrix of taxonomy and correlations (r)
# colnames(cor_spp_a1)[7] <- "Corr"
#   # renames correlation column name
# cor_spp_a1 <- cor_spp_a1[order(cor_spp_a1[,7],decreasing=TRUE),]
#   # sorts based on r
# responders_a1 <- cor_spp_a1[a1 <- abs(cor_spp_a1[,7])>0.7,]
#   # subset based on r > |0.7|
# 
# # PCoA Axis 2 Responders
# cor_spp_a2 <- cbind(EC_tax[,3],pcoaS$cproj[,2])
#   # creates matrix of taxonomy and correlations(r) axis 2
# colnames(cor_spp_a2)[7] <- "Corr"
#   # renames correlation column name
# cor_spp_a2 < -cor_spp_a2[order(cor_spp_a1[,7],decreasing=TRUE),]
#   # sorts based on r
# responders_a2 <- cor_spp_a2[a2 <- abs(cor_spp_a2[,7])>0.7,]
#   # subset based on r > |0.7|


# Other Stuff
```{r}
## PCoA Plots ##################################################################
################################################################################

WL.pcoa <- cmdscale(sampleREL.dist, eig = TRUE, k = 3) 
explainvar1 <- round(WL.pcoa$eig[1] / sum(WL.pcoa$eig), 3) * 100
explainvar2 <- round(WL.pcoa$eig[2] / sum(WL.pcoa$eig), 3) * 100
explainvar3 <- round(WL.pcoa$eig[3] / sum(WL.pcoa$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2, explainvar3)

# Define Plot Parameters
par(mar = c(5, 5, 1, 1) + 0.1)

# Define Plot Symbols
WL.pch <- rep(NA, length(design$Treatment))
for (i in 1:length(design$Treatment)){

  }

WL.pch <- rep(17, length(design$Treatment))

# Initiate Plot
plot(WL.pcoa$points[ ,1], WL.pcoa$points[ ,2], 
     # ylim = c(-0.31, 0.25), xlim = c(-0.3, 0.6), 
     xlab = paste("PCoA 1 (", explainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2, "%)", sep = ""),
     pch = WL.pch, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, 
     axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1, las = 1)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 1, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 2, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Add Points & Labels
points(WL.pcoa$points[ ,1], WL.pcoa$points[ ,2],
       pch = WL.pch, cex = 1.25, bg = "gray", col = "gray")

ordiellipse(cbind(WL.pcoa$points[ ,1], WL.pcoa$points[ ,2]),
            design$Treatment, kind="se", conf=0.95,
            lwd=2, draw = "polygon", col="gray", border = "black", label=TRUE, 
            cex=1, bty = 'n')

text(WL.pcoa$points[ ,1], WL.pcoa$points[ ,2], design$Plot)


names(which(WL.pcoa$points[, 1] > 0.3))

test <- which(WL.pcoa$points[, 1] > 0.3)

###
# Plot with odd samples removed
###
WLdataREL2 <- WLdataREL[c(WL.pcoa$points[, 1] < 0.3), ]
design2 <- design[-which(WL.pcoa$points[, 1] > 0.3), ]
sampleREL.dist2 <- vegdist(WLdataREL2, method="bray")


WL.pcoa <- cmdscale(sampleREL.dist2, eig = TRUE, k = 3) 
explainvar1 <- round(WL.pcoa$eig[1] / sum(WL.pcoa$eig), 3) * 100
explainvar2 <- round(WL.pcoa$eig[2] / sum(WL.pcoa$eig), 3) * 100
explainvar3 <- round(WL.pcoa$eig[3] / sum(WL.pcoa$eig), 3) * 100
sum.eig <- sum(explainvar1, explainvar2, explainvar3)

# Define Plot Parameters
par(mar = c(5, 5, 1, 1) + 0.1)

# Define Plot Symbols
WL.pch <- rep(NA, length(design$Treatment))
for (i in 1:length(design$Treatment)){
  
}

WL.pch <- rep(17, length(design2$Treatment))

# Initiate Plot
plot(WL.pcoa$points[ ,1], WL.pcoa$points[ ,2], 
     # ylim = c(-0.31, 0.25), xlim = c(-0.3, 0.6), 
     xlab = paste("PCoA 1 (", explainvar1, "%)", sep = ""),
     ylab = paste("PCoA 2 (", explainvar2, "%)", sep = ""),
     pch = WL.pch, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2, 
     axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1, las = 1)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=-0.02)
axis(side = 1, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 2, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 3, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
axis(side = 4, labels = F, lwd.ticks = 2, cex.axis = 1, las = 1, tck=0.01)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Add Points & Labels
points(WL.pcoa$points[ ,1], WL.pcoa$points[ ,2],
       pch = WL.pch, cex = 1.25, bg = "gray", col = "gray")

ordiellipse(cbind(WL.pcoa$points[ ,1], WL.pcoa$points[ ,2]),
            design2$Treatment, kind="se", conf=0.95,
            lwd=2, draw = "polygon", col="gray", border = "black", label=TRUE, 
            cex=1, bty = 'n')

text(WL.pcoa$points[ ,1], WL.pcoa$points[ ,2], design2$Plot)


```
# #conduct an Analysis of Similarities (ANOSIM)
# #if consulting dissimilarity ranks, use values from 50% column
# Anosim <- anosim(sampleREL.dist, grouping = site, permutations = 1000, distance = "bray")
# summary(Anosim)
# 
# #conduct a Similarity Percentages analysis
# #an overall average proportional dissimilarity between treatments is not reported, but it can be obtained by summing the values in the contribution column
# #ratio is the average proportional contribution of the respective OTU divided by its SD
# Simper <- simper(sampleREL.dist, group = site)
# summary(Simper)

